File: InterviewHub/my-react-app/src/services/categoryService.js
import axiosClient from './axiosClient';

export const categoryService = {
  getAllCategories: (params) => {
    return axiosClient.get('/categories', { params });
  },
  getCategoryById: (id) => {
    return axiosClient.get(`/categories/${id}`);
  },
  createCategory: (data) => {
    return axiosClient.post('/categories', data);
  },
  updateCategory: (id, data) => {
    return axiosClient.put(`/categories/${id}`, data);
  },
  deleteCategory: (id) => {
    return axiosClient.delete(`/categories/${id}`);
  }
};

File: Backend/src/main/java/com/InterviewHub/feature/category/CategoryService.java
package com.InterviewHub.feature.category;

import com.InterviewHub.dto.PaginationDTO;
import com.InterviewHub.feature.category.dto.CategoryResponse;
import com.InterviewHub.feature.category.dto.CreateCategoryRequest;
import com.InterviewHub.feature.category.dto.UpdateCategoryRequest;

public interface CategoryService {
    PaginationDTO<CategoryResponse> getAllCategories(int page, int size);
    CategoryResponse getCategoryById(Long id);
    CategoryResponse createCategory(CreateCategoryRequest request);
    CategoryResponse updateCategory(Long id, UpdateCategoryRequest request);
    void deleteCategory(Long id);
}
